application/event_processor/event_processor.py

```diff
@@ -237,16 +240,16 @@ def lambda_handler(event: SQSEvent, context: LambdaContext, metrics) -> None:
         return
 
     try:
-        validate_event(change_event)
-
+        service_type: ServiceType = validate_event(change_event)
+        logger.append_keys(service_type=service_type.name)
         nhs_entity = NHSEntity(change_event)
         logger.append_keys(ods_code=nhs_entity.odscode)
         logger.append_keys(org_type=nhs_entity.org_type)
         logger.append_keys(org_sub_type=nhs_entity.org_sub_type)
         metrics.set_property("ods_code", nhs_entity.odscode)
         logger.info("Created NHS Entity for processing", extra={"nhs_entity": nhs_entity})
-        event_processor = EventProcessor(nhs_entity)
-        matching_services = event_processor.get_matching_services()
+        event_processor = EventProcessor(nhs_entity, service_type)
+        matching_services = event_processor.get_matching_services(service_type)
```

Comment:
if you pass the service_type into the constructor why would you pass it again here? surely get_matching_services can use self.service_type

Reply:
Unnecessary as removed from event processor class